providers/os/resources/privatekey.go

```diff
@@ -33,7 +32,7 @@ func (r *mqlPrivatekey) id() (string, error) {
 	// TODO: use path or hash depending on initialization
 
 	file := r.GetFile()
-	if file.Error != nil {
+	if file.Error != nil || file.Data == nil {
```

Comment:
I am a bit confused by this, it implies that the Error and Data might be nil at the same time

if that is the case, we should probably split those two checks, else `return "", file.Error` might evaluate to `"", nil`

Reply:
+1, we can return a custom error if `file.Data == nil` and just let users know that no file is provided